### ISSUE TITLE
Not working on UWP apps such as MS Edge

### ISSUE THREAD

**Issue by prestyman** [benbuck/rbtray#22]
Appears to send program to system try while still leaving tab open in Taskbar.
![image](https://user-images.githubusercontent.com/6760197/54141864-bed86280-43e3-11e9-8724-d863a2af49e3.png)

**Comment by Calinou** [comment]
Does it work with other UWP applications such as Calculator?

**Comment by prestyman** [comment]
> Does RBTray work with other UWP applications such as Calculator?

No, same issue.

**Comment by benbuck** [comment]
Hi @prestyman,

I am able to reproduce this with both Edge and Calculator. Unfortunately I don't have any idea how to fix it at the moment.

-Benbuck

**Comment by prestyman** [comment]
Fair enough!  My main use case it for minimizing the browser while using a radio web player. Simple solution was to use a browser that RBTray works with, instead of Edge.

**Comment by cosmicdan** [comment]
I've found that this doesn't work with many windows anymore, not sure why. I am running as admin (I can tray minimize e.g. Task Manager) but can't minimize just about anything else, such as Explorer or VirtualBox.

**Comment by Merlin2001** [comment]
@cosmicdan : I guess you're experiencing a different problem (i.e. not related to UWP apps on Windows 10). In my experience, if you run RBTray with administrative privileges, you will only be able to minimize other admin windows (like Task Manager) with it. It's also true the other way round: If it's started with default privileges, you _can't_ minimize the Task Manager.

Unfortunately there seems to be no way to make both cases work and I don't know if @benbuck could even do something about it. (My clipboard manager app [Ditto](https://ditto-cp.sourceforge.io/) has the same limitation and works around it by showing an admin prompt whenever you try to paste into admin windows.)

**Comment by Sangatster** [comment]
<tmi>
I am a long time DM2 user.  DM2 did everything RBTray does but let you define things like mouse button and action.  No doubt had a bigger footprint than RBTray with all the options.  Progressively less compatible and with the last Windows Downdate, it started causing issues with apps, so I am switching to RBTray.
</tmi>

To prestyman: not RBTray's fault.  DM2 never worked with Metro (UWP) apps either.  Universal Windows Programs don't fully support the Windows core OS API.  Don't know if it's caused by the click event or the call that sends window to tray but whatever the case, no joy.

To benbuck:  Thanks for the touchups on this app, it is working for me much better than DM2.  I can minimize apps I haven't been able to control for years again!  Office apps even work!  If you are getting the click event then the problem has to do with the host window.  Using a spy app I wrote YEARS ago, the topmost window class is Windows.UI.Core.CoreWindow.  This is the actual app window.  If you walk the parentage, however, the UWP host app is ApplicationFrameWindow.  Post a WM_HIDE to this and Edge itself disappears, but the tasklist still looks like it's open.  The host app isn't well connected to the child so if you post a WM_HIDE, the related windows don't hide, but the main app window does.  I'm counting a total of 10 windows with Window Title or Window Class that contains 'edge' and all I have open is the browser with 1 tab.  To further complicate matters, they're not all the same PID, Other window classes are: EdgeUiInputTopWndClass (3 of those, all hidden), DummyDWMListenerWindow, 

Not a lot of time to dig around but it's an interesting problem.  I suspect if you searched for these classes and some choice keywords like controlling, hide, tray, etc., you'll find others have already done the legwork on it.

fwiw
/\/\ike \/\/elch

**Comment by benbuck** [comment]
Hi,
As @Sangatster says, this issue is more complicated that it would seem to be on the surface. Unfortunately with UWP apps Microsoft doesn't seem to be following the conventional simple window management techniques, which seems to not get along well with how RBTray is expecting windows to behave. I don't see a reasonable solution for this at the moment, but if anyone has suggestions I'm happy to get them.

Thanks,
Ben

**Comment by cpo-ux** [comment]
Doesn't work with Windows 10 (1909) Mail app. Same thing, appears to send program to system tray while still leaving tab in Taskbar.  In this case, icon in system tray is a generic one, in the attached image it's the one on the far left.

![2020-01-16 14_36_16-](https://user-images.githubusercontent.com/59980654/72564864-f0288700-386d-11ea-88a4-d53860131b26.jpg)

**Comment by mahmoudajawad** [comment]
I gave this app a shot for the sake of becoming able to run Alexa in the background without having its icon be persistent in the taskbar. Unfortunately, I came here then searching for any issues with UWP and rbtray and it seems to be the case; I tried number of win32, UWP apps and can confirm the latter doesn't work with UWP.

**Comment by benbuck** [comment]
I haven't tested this in the while, but today, running with the current RBTray and the latest Windows 10 release (19041), I am able to minimize UWP apps just fine with the right button method and the Win-Alt-Down method, and sometimes with the shift-right-click on the title bar (depending on which app).

I'm going to close this issue now since it seems to have been resolved my Microsoft.

-Benbuck

**Comment by ckuai** [comment]
I don't think this is fixed, with the Win Alt Down, it just leave the task bar button exist and a general icon in the system tray, This is exactly the problem this bug originally opened,.

I am on latest win10 2004 19041 build and rbtray v 4.11 64 bit.

**Comment by benbuck** [comment]
Hi @ckuai,

I am running the same versions you listed, and when I minimize Edge using Win-Alt-Down it works correctly. If you have any other information that could help identify what's different between your setup and mine, please let me know.

Thanks,
Benbuck

**Comment by platima** [comment]
Win 10 x64 2004 (19041.685) I am having this issue. Have not tested on 20H2 yet, but I can do on another system shortly.

Fail: Messenger, iTunes, Wacom Inkspace, Whiteboard, Your Phone
Works: Chrome, Outlook, Excel, WhatsApp, Teams, eM Client, Terminal, Debian

In this testing shift+rc title just gives context menu, rc minimise, and win+alt+down both result in the partial completion.

**Comment by benbuck** [comment]
Hi @platima,

I was able to reproduce this with the Your Phone app, but I still have no idea how to fix it. If I figure out anything I'll post it here.

Thanks,
Benbuck

**Comment by platima** [comment]
Persists in 20H2 it seems (tested on another system)

**Comment by vertigo220** [comment]
I initially thought the issue was just that RBTray had to be run as admin, since it didn't seem to be working at first until I did that, but only after searching the issues to post a suggestion to mention this in the wiki (main page) did I realize it works the other way, too. And by the way, I do think this info should be posted there, so new users aren't confused either by it seeming to not work or only work sometimes.

As for there being no solution, I can at least say I'm pretty sure that's not true. AnVir Task Manager Free, which I use for minimizing things to tray as well as many other things works on both admin and non-admin windows. It works by shift-clicking the minimize button (hence why I figured RBTray would be a nice addition, so it could be done that much easier), so maybe the use of a modifier key changes how it hooks or the requirements of the hooking somehow versus just clicking, but it definitely works, so it seems RBTray should, in theory, be able to as well. Of course, that would require figuring out how AnVir works, and it's closed-source, but maybe someone knows how to monitor calls to see what's happening in the background when that's used.

**Comment by garfieldbanks** [comment]
> As for there being no solution, I can at least say I'm pretty sure that's not true.
[I hacked on it for a bit and was able to get a user-level instance running alongside an admin-level instance by changing one of the instances to RBTray2 and using MBUTTON instead of RBUTTON.](https://github.com/benbuck/rbtray/issues/24#issuecomment-569036762)

**Comment by garfieldbanks** [comment]
> As for there being no solution, I can at least say I'm pretty sure that's not true.

[I hacked on it for a bit and was able to get a user-level instance running alongside an admin-level instance by changing one of the instances to RBTray2 and using MBUTTON instead of RBUTTON.](https://github.com/benbuck/rbtray/issues/24#issuecomment-569036762)

**Comment by Sangatster** [comment]
Interesting, maybe the issue has to do with how mouse messages are
processed by UWP apps then.
